### PR TITLE
docs: adding field descriptions to predefined image document

### PR DESCRIPTION
**Pull request by punndcoder28**
* Adds field descriptions and example usage of each field for the predefined `Image` document

### CODE REVIEW

File: docarray/documents/image.py [hunk 1]
```diff
@@ -92,10 +95,25 @@ class MultiModalDoc(BaseDoc):
...
+    )
+    embedding: Optional[AnyEmbedding] = Field(
+        description='''Embedding field is used to store tensor objects of type
+        Tensorflow, PyTorch, NumPy and Jax''',
+        example='np.zeros((3, 32, 32))',
+    )
+    bytes_: Optional[ImageBytes] = Field(
+        description='''Bytes object of the image which is an instance of
+        `ImageBytes.''',
+        example='img.url.load_bytes()',
```
Comment (JoanFM) [inline docarray/documents/image.py:115]: I do not think is a good example, I would just remove it

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docarray/documents/image.py (hunk 1 above)
Comment (punndcoder28) [reply docarray/documents/image.py]: Removed example for `bytes_`

File: docarray/documents/image.py [hunk 2]
```diff
@@ -92,10 +95,24 @@ class MultiModalDoc(BaseDoc):
...
+    )
+    tensor: Optional[ImageTensor] = Field(
+        description='''Tensor object of the image which can be specifed to one of
+    `ImageNdArray`, `ImageTorchTensor`, `ImageTensorflowTensor`.''',
+        example='img.url.load()',
+    )
+    embedding: Optional[AnyEmbedding] = Field(
+        description='''Embedding field is used to store tensor objects of type
+        Tensorflow, PyTorch, NumPy and Jax''',
+        example='np.zeros((3, 32, 32))',
```
Comment (JoanFM) [inline docarray/documents/image.py:110]: same as said for `text`. This is not the way examples for embedding should be seen. Should be a list of floats or a list of lists

File: docarray/documents/image.py [hunk 3]
```diff
@@ -92,10 +95,24 @@ class MultiModalDoc(BaseDoc):
...
-    embedding: Optional[AnyEmbedding]
-    bytes_: Optional[ImageBytes]
+    url: Optional[ImageUrl] = Field(
+        description='URL of the remote image',
+        example='https://github.com/docarray/docarray/blob/main/tests/toydata/image-data/apple.png?raw=true',
+    )
+    tensor: Optional[ImageTensor] = Field(
+        description='''Tensor object of the image which can be specifed to one of
+    `ImageNdArray`, `ImageTorchTensor`, `ImageTensorflowTensor`.''',
+        example='img.url.load()',
```
Comment (JoanFM) [inline docarray/documents/image.py:105]: let's remove this if we cannot show. Think about the example are something u can pass in the `json` and be accepted by a REST service.

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docarray/documents/image.py (hunk 3 above)
Comment (punndcoder28) [reply docarray/documents/image.py]: Updated PR with suggestions

File: docarray/documents/image.py (hunk 2 above)
Comment (punndcoder28) [reply docarray/documents/image.py]: Updated PR with suggestions

File: docarray/documents/image.py [hunk 4]
```diff
@@ -92,10 +95,21 @@ class MultiModalDoc(BaseDoc):
...
+        description='''URL to a (potentially remote) image file that needs to be
+        loaded''',
+        example='https://github.com/docarray/docarray/blob/main/tests/toydata/image-data/apple.png?raw=true',
+    )
+    tensor: Optional[ImageTensor] = Field(
+        description='Tensor object of one of the image tensors',
+    )
+    embedding: Optional[AnyEmbedding] = Field(
+        description='Store an embedding: a vector representation of the image.',
+        example='[[0, 1, 1], [1, 0, 1], [1, 1, 0]]',
```
Comment (JoanFM) [inline docarray/documents/image.py:108]:
```suggestion
        example='[1, 0, 1]',
```

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by JoanFM** [comment]
To resolve the conflicts, make sure to add None as the default value for the Optional fields.

File: docarray/documents/image.py [hunk 5]
```diff
@@ -92,10 +95,30 @@ class MultiModalDoc(BaseDoc):
     ```
     """
 
-    url: Optional[ImageUrl] = None
-    tensor: Optional[ImageTensor] = None
-    embedding: Optional[AnyEmbedding] = None
-    bytes_: Optional[ImageBytes] = None
+    url: Optional[ImageUrl] = Field(
+        description='''URL to a (potentially remote) image file that needs to be
```
Comment (JoanFM) [inline docarray/documents/image.py:99]:
```suggestion
        description='URL to a (potentially remote) image file that needs to be
```

File: docarray/documents/image.py [hunk 6]
```diff
@@ -92,10 +95,30 @@ class MultiModalDoc(BaseDoc):
...
+        loaded''',
```
Comment (JoanFM) [inline docarray/documents/image.py:100]:
```suggestion
        loaded',
```

File: docarray/documents/image.py [hunk 7]
```diff
@@ -92,10 +95,30 @@ class MultiModalDoc(BaseDoc):
...
+    embedding: Optional[AnyEmbedding] = Field(
+        description='''Embedding field is used to store tensor objects of type
+        Tensorflow, PyTorch, NumPy and Jax''',
+        example='[1, 0, 1]',
+        default=None,
+    )
+    bytes_: Optional[ImageBytes] = Field(
+        description='''Bytes object of the image which is an instance of
+        `ImageBytes.''',
+        example='img.url.load_bytes()',
```
Comment (JoanFM) [inline docarray/documents/image.py:119]: remove this example. This is not a valid one

File: docarray/documents/image.py [hunk 8]
```diff
@@ -92,10 +95,30 @@ class MultiModalDoc(BaseDoc):
...
+    tensor: Optional[ImageTensor] = Field(
+        description='''Tensor object of the image which can be specifed to one of
+    `ImageNdArray`, `ImageTorchTensor`, `ImageTensorflowTensor`.''',
+        example='img.url.load()',
+        default=None,
+    )
+    embedding: Optional[AnyEmbedding] = Field(
+        description='''Embedding field is used to store tensor objects of type
+        Tensorflow, PyTorch, NumPy and Jax''',
+        example='[1, 0, 1]',
```
Comment (JoanFM) [inline docarray/documents/image.py:113]:
```suggestion
        example=[1, 0, 1],
```

File: docarray/documents/image.py [hunk 9]
```diff
@@ -92,10 +95,30 @@ class MultiModalDoc(BaseDoc):
...
+        default=None,
+    )
+    tensor: Optional[ImageTensor] = Field(
+        description='''Tensor object of the image which can be specifed to one of
+    `ImageNdArray`, `ImageTorchTensor`, `ImageTensorflowTensor`.''',
+        example='img.url.load()',
+        default=None,
+    )
+    embedding: Optional[AnyEmbedding] = Field(
+        description='''Embedding field is used to store tensor objects of type
```
Comment (JoanFM) [inline docarray/documents/image.py:111]: please use single quotes

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by punndcoder28** [comment]
Weird, must've gotten added while rebasing the changes. Will fix the quotes

**Review by JoanFM — APPROVED** [review]

**Pull request merged**